runtime/compiler/rpc/J9Client.cpp

```diff
@@ -270,4 +273,44 @@ J9ClientStream::shutdown()
    // destructor is used instead
    }
 
+void
+J9ClientStream::markServerUnreachable(J9JITConfig *jitConfig)
+   {
+   PORT_ACCESS_FROM_JITCONFIG(jitConfig);
+   if (!_serverUnreachable)
+      {
+      _serverUnreachable = true;
+      _localCompStartTime = j9time_current_time_millis();
+      }
+   }
+
+bool
+J9ClientStream::isServerReachable(J9JITConfig *jitConfig, TR::PersistentInfo *info)
+   {
+   if (!_serverUnreachable)
+      return true;
+
+   // check server availability 1 second after the first StreamFailure
+   // if it fails again, the _pingServerInterval will be set to 10 seconds
+   // meaning the server will be pinged every 10 seconds to see if it's back up again
+   // without the 1 second delay after the first StreamFailure would cause the performance of client
+   // to suffer as client could fire off a lot of remote compilations that fails
+   PORT_ACCESS_FROM_JITCONFIG(jitConfig);
+   uint64_t current_time = j9time_current_time_millis();
+   if ((current_time - _localCompStartTime) > _pingServerInterval)
+      {
+      if (gethostbyname(info->getJITaaSServerAddress().c_str()))
```

Comment:
Is gethostbyname actually sending a ping?

Reply:
`gethostbyname` is not actually sending a ping, it just queries to see if the hostname exists. I think I need to implement an actual ping routine.

Reply:
https://github.com/eclipse/openj9/issues/6120
I opened an issue, going to change the use of `gethostbyname` to `getaddrinfo` for client openConnection soon, in a separate PR.

Reply:
A better solution, instead of pinging the server, we can try re-establishing the connection right there, we will get the same effect if it fails and the connection will be conveniently re-established if it goes through.